File: detect_changes.py
import subprocess

# Get the list of changed files in the last commit
changed_files = (
    subprocess.check_output('git diff --name-only HEAD~1 --', shell=True)
    .decode('utf-8')
    .split('\n')
    .filter(None)
)

# Debugging output
print('Changed Files:', changed_files)

# Filter out only the .swift files
swift_files = [file for file in changed_files if file.endswith('.swift')]

# Debugging output
print('Swift Files:', swift_files)

# Create a report of changed .swift files
report = f'Changed Swift Files:\n{"\n".join(swift_files)}'

# Print the report to standard output
print(report)

File: detect_changes.js
const { execSync } = require('child_process');
const fs = require('fs');

// Get the list of changed files in the last commit
const changedFiles = execSync('git diff --name-only HEAD^ HEAD')
  .toString()
  .split('\n')
  .filter(Boolean);

// Filter out only the .swift files
const swiftFiles = changedFiles.filter(file => file.endsWith('.swift'));

// Create a report of changed .swift files
const report = `Changed Swift Files:\n${swiftFiles.join('\n')}`;

// Write the report to a file
fs.writeFileSync('path/to/changes-report.txt', report);

console.log(report);
